File: src/main/java/org/cannonbank/core/Entities/Account.java
package org.cannonbank.core.Entities;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;
import javax.persistence.*;

import lombok.*;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "account")

public class Account implements java.io.Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int accountId;

	@ManyToOne
	private CategoryAccount categoryAccount;

	@ManyToOne
	private Client client;

	private String accountNumber;
	private String bic;
	private String iban;
	private float balance;

	@Temporal(TemporalType.DATE)
	private Date creationDate;

	@ManyToOne
	private AccountStatus status;

	@OneToMany(fetch = FetchType.LAZY, mappedBy="account")
	private Set<CreditCard> creditCards = new HashSet<CreditCard>(0);

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "account")
	private Set<Checkbook> checkbooks = new HashSet<Checkbook>(0);

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "accountSrc")
	private Set<Transaction> transactionsForIdS = new HashSet<Transaction>(0);

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "accountRcv")
	private Set<Transaction> transactionsForIdR = new HashSet<Transaction>(0);

	@OneToMany(fetch = FetchType.LAZY, mappedBy = "account")
	private Set<Recharge> recharges = new HashSet<Recharge>(0);

}

File: src/app/models/Account.ts
import {Client} from "./Client";
import {CategoryAccount} from "./category-account";
import {AccountStatus} from "./account-status";
import {Checkbook} from "./Checkbook";
import {CreditCard} from "./credit-card";
import {Transaction} from "./transaction";

export interface Account
{

  idAccount : number;
  categoryAccount : CategoryAccount;
  client : Client;
  accountNumber : string;
  bic : string;
  iban : string;
  balance : number;
  creationDate : Date;
  status : AccountStatus;
  creditCards : Array<CreditCard>;
  checkbook : Array<Checkbook>;
  transactionsForIdS : Array<Transaction>;
  transactionsForIdR : Array<Transaction>;

}
